src/content/guides/scaffolding.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Scaffolding
+contributors:
+  - pranshuchittora
+sort: 2
+---
+
+Setting up webpack for the first time is hard. Writing advanced configurations to optimize performance is even tougher. The `init` feature is designed to support people that want to create their own configuration or initializing other projects people create.
+
+## Creating a scaffold
+
+Before writing a `webpack-cli` scaffold, think about what you're trying to achieve. Do you want a "general" scaffold that could be used by any project or type of app? Do you want something focused, like a scaffold that writes both your `webpack.config.js` and your framework code? It's also useful to think about the user experience for your scaffold.
+
+`webpack-cli` offers an experience that is interactive and you can prompt users for questions (like, "What is your entry point?") to help customize the output accordingly.
+
+### Writing a scaffold
+
+There are many resources where you can learn how to write a scaffold, you can start from: [How do I compose a
+webpack-scaffold?](https://github.com/evenstensberg/webpack-scaffold-demo)
+
+`webpack-scaffold` is a utility suite for creating scaffolds. It contains functions that could be used to create a scaffold.
+
+### Running a scaffold
+
+A scaffold can be executed using `webpack-cli init`:
+
+```bash
+webpack-cli init <your-scaffold>
+```
+
+#### Running a scaffold locally
+
+When the scaffold package is in your local file system you should pass its path to `init`:
+
+```bash
+webpack-cli init path/to/your/scaffold
+```
+
+Or you can create a global module and symlink to the local one:
+
+- Using npm
+
+      	```bash
+      	cd path/to/my-scaffold
+      	npm link
+      	webpack-cli init my-scaffold
+      	```
+
+- Using yarn
+
+      	```bash
+      	cd path/to/my-scaffold
+      	yarn link
+      	webpack-cli init my-scaffold
+
+#### Running a scaffold from npm
+
+If the package is in npm, its name must begin with `webpack-scaffold` and can be used running:
+
+```bash
+webpack-cli init webpack-scaffold-yourpackage
+```
+
+## API
+
+To create a `scaffold`, you must create a [`yeoman-generator`](http://yeoman.io/authoring/). Because of that, you can optionally extend your generator to include methods from the [Yeoman API](http://yeoman.io/learning/). It's worth noting that we support all the properties of a regular webpack configuration. In order for us to do this, there's a thing you need to remember:
+
+> Objects are made using strings, while strings are made using double strings. This means that in order for you to create a string, you have to wrap it inside another string for us to validate it correctly.
+
+### Required
+
+- [opts.env.configuration](#optsenvconfiguration-required)
+- [opts.env.configuration.myObj](#optsenvconfigurationmyObj-required)
+- [myObj.webpackOptions](#myObjwebpackOptions-required)
```

Comment:
Using `config` is more intuitive instead of `myObj` IMO. We can have it changed everywhere.
pinging @evenstensberg for final call.

Reply:
I think it's might get confusing  as `opts.env.configuration.myObj` would become `opts.env.configuration.config`

Reply:
In another PR Even suggested me not to change `myObj` to `myConfig` because it was confusing as sais Pranshu.
But I agree with Rishabh that it's not very intuitive, IMO would be nice to have a more meaningful name that gives the idea that it will become a webpack configuration file

Reply:
Yup, makes sense, what about `customConfig` or something like that.

Reply:
I would also consider changing `opts.env.configuration` to `opts.env.configurations`

Reply:
regarding configurations, yes. Regards to myObj is bc it sounds singular, not plural. You can nest the boostrap object however you want, needs to be emphasized if we're changing that.

That's all paper engineering. Submit an issue and let's go through that separately as in issue -> assignee -> pr -> release.